lightning/src/ln/channelmanager.rs

```diff
@@ -2815,10 +2815,10 @@ enum NotifyOption {
 /// We allow callers to either always notify by constructing with `notify_on_drop` or choose to
 /// notify or not based on whether relevant changes have been made, providing a closure to
 /// `optionally_notify` which returns a `NotifyOption`.
```

Comment:
Background question: it seems that in some instances, the `persist_check` isn't just checking, but also making the actual changes to the chan mgr state. Is that fine, or indicative of the `PersistenceNotifierGuard` not fully matching its desired use cases?

Reply:
Will take a closer look at current uses of this. Still it feels to me that the general idea of a scope with a guard is to do the manipulation inside that scope, not deeper nested in the `should_persist` callback.